extension/sql/timescale-prometheus.sql

```diff
@@ -159,4 +159,67 @@ AS $func$
     VALUES ('timescale_prometheus_' || meta_key,meta_value, send_telemetry)
     ON CONFLICT (key) DO UPDATE SET value = EXCLUDED
 $func$
-LANGUAGE SQL VOLATILE SECURITY DEFINER;
\ No newline at end of file
+LANGUAGE SQL VOLATILE SECURITY DEFINER;
+
+----------------------------  aggregation functions ----------------------------
+
+CREATE FUNCTION @extschema@.gp_d_t(state internal, lowest_time timestamptz,
+    greatest_time timestamptz, step_size bigint, window_size bigint,
+    timev timestamptz, val double precision
+    )
+RETURNS internal AS '$libdir/timescale_prometheus_extra', 'gapfill_delta_transition'
+LANGUAGE C STABLE PARALLEL SAFE;
+
+CREATE FUNCTION @extschema@.gp_r_t(state internal, lowest_time timestamptz,
+    greatest_time timestamptz, step_size bigint, window_size bigint,
+    timev timestamptz, val double precision
+    )
+RETURNS internal AS '$libdir/timescale_prometheus_extra', 'gapfill_rate_transition'
+LANGUAGE C STABLE PARALLEL SAFE;
+
+CREATE FUNCTION @extschema@.gp_i_t(state internal, lowest_time timestamptz,
+    greatest_time timestamptz, step_size bigint, window_size bigint,
+    timev timestamptz, val double precision
+    )
+RETURNS internal AS '$libdir/timescale_prometheus_extra', 'gapfill_increase_transition'
+LANGUAGE C STABLE PARALLEL SAFE;
+
+CREATE FUNCTION @extschema@.gp_d_f(state internal)
+RETURNS DOUBLE PRECISION[]
+AS '$libdir/timescale_prometheus_extra', 'gapfill_delta_final' LANGUAGE C;
+
+CREATE AGGREGATE @extschema@.gapfill_delta(lowest_time timestamptz,
+    greatest_time TIMESTAMPTZ,
+    step_size BIGINT,
+    window_size BIGINT,
```

Comment:
in this public api may want to call it range_size to be compatible with prom terminology